lib/widgets/info_card.dart

```diff
@@ -6,104 +6,116 @@ import '../resources/theme.dart';
 class InfoCard<T> extends StatelessWidget {
   const InfoCard(
       {@required this.future,
-      @required this.builder,
-      this.onShowMore,
-      this.title,
-      this.showMoreButtonKey,
-      this.padding});
+        @required this.builder,
+        this.onShowMore,
+        this.showIfEmpty = false,
+        this.title,
+        this.showMoreButtonKey,
+        this.padding});
 
   final Function onShowMore;
   final Future<T> future;
   final Widget Function(T) builder;
   final String title;
   final ValueKey<String> showMoreButtonKey;
   final EdgeInsetsGeometry padding;
+  final bool showIfEmpty;
 
   @override
   Widget build(final BuildContext context) {
-    return Padding(
+
+    return FutureBuilder(
+        future: future,
+        builder: (final context, final snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            if (snapshot.hasData) {
+              if ((snapshot.data is Map ||
+                  snapshot.data is Iterable) &&
+                  snapshot.data.isEmpty) {
+                return showIfEmpty ? cardWrapper(context, noneYet(context)) : const SizedBox();
+              }
+              return cardWrapper(context, builder(snapshot.data));
+            } else {
+              return showIfEmpty ? cardWrapper(context, noneYet(context)) : const SizedBox();
+            }
```

Comment:
This can be a single if

Reply:
got rid of this complicated double-if statement and replace it with a more readable construction
I left some comments for better code understanding (this comment has been solved on the new PR as well)